src/PrimeTower.cpp

```diff
@@ -247,16 +249,118 @@ void PrimeTower::preWipe(const SliceDataStorage& storage, LayerPlan& gcode_layer
         GCodePath& toward_middle = gcode_layer.addTravel(middle);
         toward_middle.perform_z_hop = true;
         gcode_layer.forceNewPathStart();
-        GCodePath& toward_wipe_start = gcode_layer.addTravel_simple(start);
-        toward_wipe_start.perform_z_hop = false;
-        toward_wipe_start.retract = true;
+
+        if (purge_volume > 0)
+        {
+            // Find out how much purging needs to be done.
+            const coord_t purge_move_length = vSize(middle - prime_start);
+            const unsigned int line_width = gcode_layer.configs_storage.prime_tower_config_per_extruder[extruder_nr].getLineWidth();
+            const double layer_height_mm = (layer_nr == 0) ? train.getSettingInMillimeters("layer_height_0") : train.getSettingInMillimeters("layer_height");
+            const double normal_volume = INT2MM(INT2MM(purge_move_length * line_width)) * layer_height_mm; // Volume extruded on the "normal" move
+            float purge_flow = purge_volume / normal_volume;
+
+            const GCodePathConfig *current_gcode_path_config = &gcode_layer.configs_storage.prime_tower_config_per_extruder[extruder_nr];
+
+            const coord_t purge_move_length_mm = INT2MM(purge_move_length);
+            const double purge_move_time = purge_move_length_mm / current_gcode_path_config->getSpeed();
+            const double purge_extrusion_speed_mm3_per_sec = purge_volume / purge_move_time;
+            const double max_possible_extursion_speed_mm3_per_sec = 3.0;
+
+            const double speed = current_gcode_path_config->getSpeed();
+            double extrusion_speed_factor = 1.0;
+
+            if (purge_extrusion_speed_mm3_per_sec > max_possible_extursion_speed_mm3_per_sec)
+            {
+                // compensate the travel speed for the large extrusion amount
+                const double min_time_needed_for_extrusion = purge_volume / max_possible_extursion_speed_mm3_per_sec;
+#ifndef NDEBUG
```

Comment:
`NDEBUG` is defined when we are building a `RELEASE` version. This is the macro which enables/disables the `assert()`s. In a `DEBUG` build, `NDEBUG` is *not* defined and `assert` will work; in a `RELEASE` build, things happen in the opposite way.